devicetypes/heltun/heltun-tps05-switch.src/heltun-tps05-switch.groovy

```diff
@@ -0,0 +1,680 @@
+	def roomTemperature = 1
+	def humidity = 5
+	def illuminance = 3
+	def localScale = getTemperatureScale() //HubScale
+	def deviceScale = (cmd.scale == 1) ? "F" : "C" //DeviceScale
+	def child = childDevices?.find {channelNumber(it.deviceNetworkId) == 1 }
+	if (roomTemperature == cmd.sensorType) {
+		def deviceTemp = cmd.scaledSensorValue
+		def scaledTemp = (deviceScale == localScale) ? deviceTemp : (deviceScale == "F" ? roundC(fahrenheitToCelsius(deviceTemp)) : celsiusToFahrenheit(deviceTemp).toDouble().round(0).toInteger())
+		map.name = "temperature"
+		map.value = scaledTemp
+		map.unit = localScale
+		sendEvent(map)
+	} else if (humidity == cmd.sensorType) {
+		map.name = "humidity"
+		map.value = cmd.scaledSensorValue.toInteger()
+		map.unit = "%"
+		sendEvent(map)
+	} else if (illuminance == cmd.sensorType) {
+		map.name = "illuminance"
+		map.value = cmd.scaledSensorValue
+		sendEvent(map)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	def parameter = parameterMap().find( {it.paramNum == cmd.parameterNumber } ).name
+	if (state."$parameter".value == cmd.scaledConfigurationValue){
+		state."$parameter".state = "configured"
+	}
+	else {
+		state."$parameter".state = "error"
+	}
+	configParam()
+}
+
+def updated() {
+	if (childDevices && device.label != state.oldLabel) {
+		childDevices.each {
+			def newLabel = getChildName(channelNumber(it.deviceNetworkId))
+			it.setLabel(newLabel)
+		}
+		state.oldLabel = device.label
+	}
+	initialize()
+}
+
+def initialize() {
+	runIn(3, "checkParam")
+}
+
+def installed() {
+	def numberOfButtons = 5
+	state.numberOfButtons = numberOfButtons
+	state.oldLabel = device.label
+	def existingChildren = getChildDevices()
+	for (i in 1..numberOfButtons) {
+		def buttonNetworkId = "${device.deviceNetworkId}:${i+2*numberOfButtons}"
+		def relayNetworkId = "${device.deviceNetworkId}:${i+numberOfButtons}"
+		def backlightNetworkId = "${device.deviceNetworkId}:${i}"
+		def childRelayExists = (existingChildren.find {child -> child.getDeviceNetworkId() == relayNetworkId} != NULL)
+		def childButtonExists = (existingChildren.find {child -> child.getDeviceNetworkId() == buttonNetworkId} != NULL)
+		def childBacklightExists = (existingChildren.find {child -> child.getDeviceNetworkId() == backlightNetworkId} != NULL)
+		if (!childBacklightExists ) {
+			def child = addChildDevice("Heltun Child Backlight", backlightNetworkId, device.hubId, [completedSetup: true, label: getChildName(i), isComponent: false])
+		}
+		if (!childRelayExists) {
+			addChildDevice("HELTUN", "Heltun Child Relay", relayNetworkId, device.hubId,[completedSetup: true, label: getChildName(i+numberOfButtons), isComponent: false])
+		}
+		if (!childButtonExists ) {
+			def child = addChildDevice("smartthings", "Child Button", buttonNetworkId, device.hubId, [completedSetup: true, label: getChildName(i+2*numberOfButtons), isComponent: true, componentName: "button$i", componentLabel: "Button ${i}"])
```

Comment:
why are you creating variable `child` if you're not referring to it in any way later?